rust/src/array.rs

```diff
@@ -0,0 +1,412 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::convert::From;
+use std::iter::Iterator;
+use std::mem;
+use std::rc::Rc;
+use std::str;
+use std::string::String;
+
+use super::bitmap::Bitmap;
+use super::error::*;
+use super::memory::*;
+
+use bytes::{Bytes, BytesMut, BufMut};
+use libc;
+
+pub enum ArrayData {
+    Boolean(*const bool),
+    Float32(*const f32),
+    Float64(*const f64),
+    Int8(*const i8),
+    Int16(*const i16),
+    Int32(*const i32),
+    Int64(*const i64),
+    UInt8(*const u8),
+    UInt16(*const u16),
+    UInt32(*const u32),
+    UInt64(*const u64),
+    Utf8(ListData),
+    Struct(Vec<Rc<Array>>)
```

Comment:
Should be `Field` instead of `Array`.

Reply:
We have Schema/Field for defining types including structs (I was wrong about this in our slack chat earlier). I added a test with an example for defining a struct.

```
        let _person = Schema::new(vec![
            Field::new("first_name", DataType::Utf8, false),
            Field::new("last_name", DataType::Utf8, false),
            Field::new("address", DataType::Struct(vec![
                Field::new("street", DataType::Utf8, false),
                Field::new("zip", DataType::UInt16, false),
            ]), false),
        ]);
```